### PR TITLE
Update docker script

### PR DESCRIPTION
**Pull request by venkateshamatam**
# Description
Update docker script to send events on first run and create a new logs dir and mount it to the container.

# Testing
Tested the script by verifying siglens.log in the logs dir and also the mounted volume inside /logs/ in the docker container. Tested the sendevents functionality by running the script multiple times to verify first run and subsequent runs functionality. Added steps to exit in case of errors.
# Checklist:
Before marking your pull request as ready for review, complete the following.

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: install_with_docker.sh [hunk 1]
```diff
@@ -228,6 +231,33 @@ check_ports
...
+    fi
+    tar -xvf 2kevents.json.tar.gz || print_error_and_exit "Failed to extract 2kevents.json.tar.gz"
+    for i in $(seq 1 20)
+    do
+        curl -s http://localhost:8081/elastic/_bulk --data-binary "@2kevents.json" -o res.txt
+        if [ $? -ne 0 ]; then
+            print_error_and_exit "Failed to send sample log dataset"
+        fi
+    done
+    print_success_message "\n Sample log dataset sent successfully"
```
Comment (nkunal) [inline install_with_docker.sh:257]: better to put the content of this if condition in a function, easier to read and follow.

File: install_with_docker.sh [hunk 2]
```diff
@@ -228,6 +231,33 @@ check_ports
...
 UI_PORT=${PORT} WORK_DIR="$(pwd)" SIGLENS_VERSION=${SIGLENS_VERSION} docker-compose -f ./docker-compose.yml up -d || print_error_and_exit "Failed to start Docker Compose"
 UI_PORT=${PORT} WORK_DIR="$(pwd)" SIGLENS_VERSION=${SIGLENS_VERSION} docker-compose logs -t --tail 20 >> dclogs.txt
+sample_log_dataset_status=$(curl -s -o /dev/null -I -X HEAD -w "%{http_code}" http://localhost:5122/elastic/sample-log-dataset)
+
+if [ "$sample_log_dataset_status" -eq 200 ]; then
+    FIRST_RUN=false
+elif [ "$sample_log_dataset_status" -eq 404 ]; then
+    FIRST_RUN=true
+else
+    print_error_and_exit "Failed to check if sample data is present. Please try again"
```
[the remark this answers is not in the document]
Comment (venkateshamatam) [reply install_with_docker.sh]: done. added non-exit error message and set first_run=true

File: install_with_docker.sh (hunk 1 above)
Comment (venkateshamatam) [reply install_with_docker.sh]: moved this to send_events function

File: install_with_docker.sh [hunk 3]
```diff
@@ -225,9 +228,41 @@ check_ports() {
...
+if [ "$sample_log_dataset_status" -eq 200 ]; then
+    FIRST_RUN=false
+elif [ "$sample_log_dataset_status" -eq 404 ]; then
+    FIRST_RUN=true
+else
+    echo "Failed to check sample log dataset status"
+    FIRST_RUN=true
+fi
+
+send_events
```
Comment (nkunal) [inline install_with_docker.sh:265]: do the `if $FIRST_RUN;` check here and if true only then call send_events.
send_events should just send_events and dont have to worry about whether it is a first run or not

**Review by nkunal — APPROVED** [review]

**Pull request merged**